### PR TITLE
NEW: Test for custom Topic properties different than default

### PR DESCRIPTION
**Pull request by asanzgom**
closes https://issues.redhat.com/browse/MGDSTRM-9533

### CODE REVIEW

**Review by Frawless — APPROVED** [review]

File: lib/topic.ts [hunk 1]
```diff
@@ -1,25 +1,61 @@
...
   await expect(page.getByText(kafkaName)).toHaveCount(1);
-  await waitForKafkaReady(page, kafkaName);
   await page.locator('a', { hasText: kafkaName }).click();
   await expect(await page.locator('button[aria-label="Topics"]')).toHaveCount(1);
   // data-testid=pageKafka-tabTopics
   await page.locator('button[aria-label="Topics"]').click();
 };
 
-export const createKafkaTopic = async function (page: Page, name: string) {
+export const createKafkaTopic = async function (page: Page, name: string, defaultProperties: boolean) {
```
Comment (mmusil) [inline lib/topic.ts:13]: Could you make this new parameter optional with a default value `false`?

File: tests/rhosak/kas_with_instance.spec.ts [hunk 2]
```diff
@@ -219,27 +216,55 @@ test('check Topic does not exist and create and delete', async ({ page }) => {
...
   await deleteKafkaTopic(page, testTopicName);
 });
 
 // test_4kafka.py test_kafka_try_create_topic_with_same_name
 test('test kafka try create topic with same name', async ({ page }) => {
-  await waitForKafkaReady(page, testInstanceName);
   await page.locator('a', { hasText: `${testInstanceName}` }).click();
   await page.locator('button[aria-label="Topics"]').click();
-  await createKafkaTopic(page, testTopicName);
+  await createKafkaTopic(page, testTopicName, true);
```
Comment (mmusil) [inline tests/rhosak/kas_with_instance.spec.ts:227]: With the default value, you would not have to update all occurrences.

File: tests/rhosak/kas_with_instance.spec.ts [hunk 3]
```diff
@@ -219,27 +216,55 @@ test('check Topic does not exist and create and delete', async ({ page }) => {
...
+  await page.locator('button', { hasText: 'Delete topic' }).click();
+  await page.getByLabel('Type DELETE to confirm:').fill('DELETE');
+  await page.locator('footer').locator('button', { hasText: 'Delete' }).click();
+});
+
 // test_4kafka.py test_edit_topic_properties_after_creation
 test('edit topic properties after creation', async ({ page }) => {
   await navigateToKafkaTopicsList(page, testInstanceName);
-  await createKafkaTopic(page, testTopicName);
+  await createKafkaTopic(page, testTopicName, true);
```
Comment (mmusil) [inline tests/rhosak/kas_with_instance.spec.ts:267]: With the default value, you would not have to update all occurrences.

File: tests/rhosak/kas_with_instance.spec.ts [hunk 4]
```diff
@@ -219,27 +216,55 @@ test('check Topic does not exist and create and delete', async ({ page }) => {
   // expecting not to find topic row
   await expect(page.locator('td', { hasText: `${testTopicName}` })).toHaveCount(0);
 
-  await createKafkaTopic(page, testTopicName);
+  await createKafkaTopic(page, testTopicName, true);
```
Comment (mmusil) [inline tests/rhosak/kas_with_instance.spec.ts:219]: With the default value, you would not have to update all occurrences.

File: tests/rhosak/kas_messaging.spec.ts [hunk 5]
```diff
@@ -72,7 +72,7 @@ test.beforeEach(async ({ page }) => {
   await expect(page.getByText('Create topic')).toHaveCount(1);
   await expect(page.getByText('Loading content')).toHaveCount(0);
   if ((await page.locator('a', { hasText: testTopicName }).count()) === 0) {
-    await createKafkaTopic(page, testTopicName);
+    await createKafkaTopic(page, testTopicName, true);
```
Comment (mmusil) [inline tests/rhosak/kas_messaging.spec.ts:75]: With the default value, you would not have to update all occurrences.

File: tests/rhosak/kas_with_instance.spec.ts [hunk 6]
```diff
@@ -219,27 +216,55 @@ test('check Topic does not exist and create and delete', async ({ page }) => {
...
+  await expect(rs).not.toMatch('Unlimited');
+
+  const cp = await page.getByLabel('Cleanup policy').getAttribute('value');
+  await expect(cp).not.toMatch('delete');
+
+  // Topic CleanUp
+  await page.locator('button', { hasText: 'Delete topic' }).click();
+  await page.getByLabel('Type DELETE to confirm:').fill('DELETE');
+  await page.locator('footer').locator('button', { hasText: 'Delete' }).click();
+});
```
Comment (mmusil) [inline tests/rhosak/kas_with_instance.spec.ts:262]: Why are you not testing whether the properties you have set are there and testing just if properties are other than default?

**Review by mmusil — COMMENTED** [review]
Hmmmm, you merged the PR while I was reviewing. 😅

File: tests/rhosak/kas_with_instance.spec.ts (hunk 6 above)
Comment (mmusil) [reply tests/rhosak/kas_with_instance.spec.ts:262]: Plus duplicating code for topic deletion, instead of using `deleteKafkaTopic`.

Comment (asanzgom) [reply tests/rhosak/kas_with_instance.spec.ts:262]: this is because at the bottom of the page there is a Delete Topic button that can be tested on the go

Comment (asanzgom) [reply tests/rhosak/kas_with_instance.spec.ts:262]: I tested regression and with True parameter these tests work also if you don't specify true or false that is an error 
image
translated: 3 arguments were expected but found 2

Comment (mmusil) [reply tests/rhosak/kas_with_instance.spec.ts:262]: Yeah, but you did not set a default value to `defaultProperties` that would be used should you not pass anything as the 3rd argument.

**Pull request merged**